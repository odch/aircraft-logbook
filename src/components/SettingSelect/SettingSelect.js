import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'
import ListItemText from '@material-ui/core/ListItemText'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import CircularProgress from '@material-ui/core/CircularProgress'
import Select from '@material-ui/core/Select'
import MenuItem from '@material-ui/core/MenuItem'
import ListItem from '@material-ui/core/ListItem'

const styles = {
  loadingIndicator: {
    verticalAlign: 'middle',
    height: '0.85em',
    marginRight: '0.7em'
  }
}

const SettingSelect = ({
  label,
  value,
  options,
  submitting,
  onChange,
  classes
}) => (
  <ListItem disableGutters>
    <ListItemText primary={label} />
    <ListItemSecondaryAction>
      {submitting && (
        <CircularProgress size={16} className={classes.loadingIndicator} />
      )}
      <Select
        value={value}
        onChange={e => onChange(e.target.value)}
        disabled={submitting}
      >
        {options.map(option => (
          <MenuItem key={option.value} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
      </Select>
    </ListItemSecondaryAction>
  </ListItem>
)

SettingSelect.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.any,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.any.isRequired,
      label: PropTypes.string.isRequired
    })
  ).isRequired,
  submitting: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired
}

export default withStyles(styles)(SettingSelect)
