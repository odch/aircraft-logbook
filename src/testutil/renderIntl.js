import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
