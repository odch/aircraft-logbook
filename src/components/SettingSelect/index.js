import SettingSelect from './SettingSelect'

export default SettingSelect
