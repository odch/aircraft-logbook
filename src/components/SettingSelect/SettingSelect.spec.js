import React from 'react'
import renderIntl from '../../testutil/renderIntl'
import SettingSelect from './SettingSelect'

describe('components', () => {
  describe('SettingSelect', () => {
    it('renders correctly without value', () => {
      const tree = renderIntl(
        <SettingSelect
          label="My test select"
          options={[
            { value: 'value1', label: 'Value 1' },
            { value: 'value2', label: 'Value 2' },
            { value: 'value3', label: 'Value 3' }
          ]}
          onChange={() => {}}
        />
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders correctly with value', () => {
      const tree = renderIntl(
        <SettingSelect
          label="My test select"
          value="value2"
          options={[
            { value: 'value1', label: 'Value 1' },
            { value: 'value2', label: 'Value 2' },
            { value: 'value3', label: 'Value 3' }
          ]}
          onChange={() => {}}
        />
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders readonly with loading indicator when submitting', () => {
      const tree = renderIntl(
        <SettingSelect
          label="My test select"
          value="value2"
          options={[
            { value: 'value1', label: 'Value 1' },
            { value: 'value2', label: 'Value 2' },
            { value: 'value3', label: 'Value 3' }
          ]}
          onChange={() => {}}
          submitting
        />
      ).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
