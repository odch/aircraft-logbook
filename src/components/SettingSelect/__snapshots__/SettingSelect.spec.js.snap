// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SettingSelect renders correctly with value 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test select
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Value 2
      </div>
      <input
        type="hidden"
        value="value2"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</li>
`;

exports[`components SettingSelect renders correctly without value 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test select
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        type="hidden"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</li>
`;

exports[`components SettingSelect renders readonly with loading indicator when submitting 1`] = `
<li
  className="MuiListItem-container"
>
  <div
    className="MuiListItem-root MuiListItem-secondaryAction"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        My test select
      </span>
    </div>
  </div>
  <div
    className="MuiListItemSecondaryAction-root"
  >
    <div
      className="MuiCircularProgress-root SettingSelect-loadingIndicator-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 16,
          "width": 16,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline Mui-disabled Mui-disabled"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled MuiInputBase-inputSelect Mui-disabled"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={null}
      >
        Value 2
      </div>
      <input
        type="hidden"
        value="value2"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</li>
`;
